keymaster/ipc/optee_keymaster_ipc.cpp

```diff
@@ -54,6 +44,12 @@ int optee_keymaster_connect(void) {
         return false;
     }
 
+    res = TEEC_InitializeContext(NULL, &ctx);
+    if (res != TEEC_SUCCESS) {
+        ALOGE("TEEC_InitializeContext failed with code 0x%x", res);
+        return (int)res;
```

Comment:
`TEEC_Result` is an `uint32_t` (where the upper bits are used), so casting to an int is probably not a good idea. I can see that this was like that in the old code as well. But I think it would be better to let this function return uint32_t if possible. Otherwise we should transform the return values to some other values.

Reply:
yes I agree and I have already future changes for that but the goal of the commit is to "just" realign the code with trusty.

So I did not want (yet) change this part and keep the current code:
https://github.com/linaro-swg/kmgk/blob/master/keymaster/ipc/optee_keymaster_ipc.cpp#L41